### ISSUE TITLE
HitBTC - LongPingFair/ShortPingFair inverted?

### ISSUE THREAD

**Issue by Zachar543** [ctubio/Krypto-trading-bot#404]
When using LongPingFair:
The value of "BuyPing" and "SellPing" are correctly displaying the highest margin Ping, but when the trade goes through, the pong is assigned to the lowest margin ping at the time.

When using ShortPingFair:
The value of "BuyPing" and "SellPing" are correctly displaying the lowest margin ping, but when the trade goes through, the pong is assigned to the highest margin ping at the time.

Basically, the pongAt setting appears to be inverted in the "Pong" assignment code. I've been seeing this issue for at least a week (I was 51 revisions behind), but it appears to still be an issue after updating today.

Here's a few settings I'm using:
Mode: Top (happens with Join as well)
Safety: Boomerang
PongAt: Any of them; including both Fair and Aggressive

Has anyone else seen this?

**Comment by ctubio** [comment]
thanks for the report'' i never experiencied this, i found the pongAt always working as expected, what currency are you using in hitbtc?

**Comment by Zachar543** [comment]
I'm currently using the BTC/USD pair on HitBTC.

Here's an example from the last hour:
![image](https://user-images.githubusercontent.com/2058925/32976573-73bce6e2-cbe7-11e7-9f45-c71001648f8f.png)

In that example: pongAt was set to ShortPingFair, BuyPing said "7,554", and there were more than 20 Buy pings of 0.01 quantity between $7,504.22 and $7,554 that would have been lower margin.

Assuming I'm understanding correctly, shouldn't the Buy Ping tied to this Pong in the table be which ever Ping is closest to the current price, or at least one that is much closer to the ping that is being displayed by the header "BuyPing:" value?

**Comment by Zachar543** [comment]
I just noticed something interesting that may be related:
The top line of that screenshot shows a 0.0000 qtyPong that appeared at the same instant as the second line. I'm not sure of the logic behind choosing the correct line, but is it possible the qtyPong of 0.0100 was actually rounded up from something like 0.00999999, and the pong was assigned to this line because it better matched the trade quantity?

**Comment by ctubio** [comment]
a pong of 0.0000 means 0.0000001 or sumshit but the precision cut the decimals to 4, this happens when a matching ping is fullfilled and the rest of the pong quantity that overflows the previous ping goes to match a next ping.

this happens because oviously you dont buy and sell always the same quantity and a single pong can go to match multiple pings
